### ISSUE TITLE
QOL features for pftool

### ISSUE THREAD

**Issue by EinarElen** [LDMX-Software/pflib#61]
Some parts of working with pftool are a bit slow to use and could benefit from being a bit more automatable, even if they are less effort than e.g. the scans. 

- A general "run this before we get beam" task 
- Setting a parameter on all ROC halves to the same value 
- Setting the LED bias on all connections at the same time 
- probably more things to add that I can't quite remember at this moment 

Most of these can of course be done manually or, ideally, with a config file but when you are working on it directly, these are things that we have found to save us a bunch of time. I've written these features mostly already but in a rather hacky/on-demand way which isn't ideal and would diverge from main much too fast, so I'm making this issue to collect them and do it more properly.

**Comment by EinarElen** [comment]
Some updates of whats in here as of 
b9d11fba3384136396f473f1c463e363575d8bcd 

- [x] Factor out certain parts of the code as functions so that they can be reused in other places without excessive quantities of copy/paste code snippets. See e.g. 865ab7839b0b05b91f30c2274ba28854ad6401a4
- [x] Add versions of such functions that don't ask the user for input. See e.g. f2f8ee0ded12a0d3d4f4873931de0c173cdbe3c4 
- [x] Moved some function static variables, hardcoded constants, or variables that were being requested multiple times into static variables in wrapper functions. These are guaranteed to be initialized the first time the function is called which means the user will only be asked for input once throughout the duration of the program. It's still a global variable even if its protected by the scope of the function though... Useful for things that aren't likely to need changing within one session (like the number of channels per elink), although if you would need to you could always just write the BaseMenu::readline_foo call instead of using these. See e.g. 2ce646421a6e2fa881e44f72045e369cea9fe4c7
- [x] Generally, factor out code that takes up a lot of real estate (LOC or mental) that aren't directly relevant to the functions that they are part of. Typically, this is things like sprintfing together a filename in a buffer in the middle of a function, see 478f8ea1bc3015bba28f2201728036769907ffc4
- [x] Add versions of features like POKE_PARAM that are applied to all halves of the connected ROCs and/or all channels of the connected ROCs. Note: There is a slight difference in how you use these. Instead of providing a page like DIGITAL_HALF_0, you only provide the template part (e.g. DIGITAL_HALF_). This isn't super intuitive, but its enough for what we need right now
- [x] Add TOA to the csv output of the charge injection scans 
- [x] Functions for setting bias to the same value on all connectors on all elinks. Currently, these functions will always run the bias initialization functions before doing anything else. This is quite a bit redundant some of the time and could definitely be improved
- [x] Split the existing charge injection scan into distinct components that can be reused (especially for the morning calibration routine). Instead of one long function, the "SCANCHARGE" command now does 
  - Create the output CSV file with the corresponding header 
  - Run preparation for charge injection (disable high/lowrange, enable IntCtest, set L1Offset) 
  - Scan over the parameter in question (e.g. CALIB_DAC). This part is then further subdivided into steps like enabling/disabling charge injection on one channel per elink...
  - Run teardown for charge injection (disable IntCtest)
- [x] Increase the verbosity in general. For the morning calibration and beam prepration run, its incredibly useful to be able to go back and see what was actually ran from stdout. It is probably too verbose at the moment and ideally, this would be toggleable in some way...
Probably more that I've forgotten to mention here... 
- [x] Add a nightmare abomination that I'm deeply ashamed of 4e74d1175521146b8d42fecd839d92c0aba8783c but which at least keeps the horrific parts stuck in one place 
- [x] Add a new task which automatically configures and performs the runs we need for taking calibration data. This includes 1 pedestal run, a couple of charge injection scans over CALIB_DAC (1 coarse grained scan over LOWRANGE, 1 fine grained scan over the ADC region of HIGHRANGE, and a coarse grained scan over the TOT region of HIGHRANGE), and a couple of LED runs with different LED bias 
- [x] Add a new tasks that runs the configuration steps we want to be sure are always done before we start taking beam if anyone has been doing anything else with the system. This includes loading the current version of the config files for each ROC, enabling DMA, printing DAQ status etc...

**Comment by EinarElen** [comment]
Updates as of a4ced6a5887773c72e8b012de41554ebaa5e5c77 
- [x] Config files for v2 and v3 
- [x] Automatic ELINK alignment procedure in tasks/
- [ ] Add a task that just takes some pedestal data and displays it with some (possibly bogus) statistics, useful for testing changes 
- [ ] A task for semi-automatically aligning preamp settings (REF_DAC_INV) 
- [ ] A task for interactively tuning pedestals channel by channel (DACB, SIGN_DAC)
- [ ] More metadata in default filenames 
- [ ] Store header check data in a silly datastructure rather than just outputting it to stdout 
- [ ] Make pflib compile without Rogue or uHAL, which makes it possibly to develop without access to (this was just adding one or two additional ugly #ifdefs) 

... Again probably more that I've forgotten
